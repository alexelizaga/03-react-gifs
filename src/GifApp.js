import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifApp = ({ defaultCategories }) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        { 
          categories.map((category, index) => (
            <GifGrid key={category + index} category={category} />
          ))
        }
      </ol>
    </>
  )
}

GifApp.propTypes = {
  defaultCategories: PropTypes.arrayOf(String),
}

GifApp.defaultProps = {
  defaultCategories: ['One Punch'],
};
