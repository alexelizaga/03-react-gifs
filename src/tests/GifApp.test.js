import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import { GifApp } from '../GifApp';

describe('Test <GifApp />', () => {

  test('should render <GifApp /> correctly', () => {
    const tree = renderer
      .create(<GifApp />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should show a categories list', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    render( <GifApp defaultCategories={categories} /> );

    const GifGridsTitle = screen.getAllByTestId('GifGridTitle');
    expect(GifGridsTitle.length).toBe(categories.length);

    const GifGrids = screen.getAllByTestId('GifGrid');
    expect(GifGrids.length).toBe(categories.length);
  });

});