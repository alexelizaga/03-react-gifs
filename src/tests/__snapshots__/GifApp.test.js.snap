// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifApp /> should render <GifApp /> correctly 1`] = `
Array [
  <h2>
    GifApp
  </h2>,
  <form
    data-testid="AddCategoryForm"
    onSubmit={[Function]}
  >
    <input
      data-testid="AddCategoryInput"
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>,
  <hr />,
  <ol>
    <div
      data-testid="GifGrid"
    >
      <h3
        className="animate__animated animate__fadeIn"
        data-testid="GifGridTitle"
      >
        One Punch
      </h3>
      <p
        className="animate__animated animate__flash"
        data-testid="GifGridLoading"
      >
        Loading...
      </p>
      <div
        className="card-grid"
      />
    </div>
  </ol>,
]
`;
